### DISCUSSION TITLE
Why does creating a GPUTexture require a format field?

### DISCUSSION THREAD

**Discussion by juj** [gpuweb/gpuweb#1903] [Q&A] [answered]
Creating a GPUTexture with device.createView() requires a `format` field in its descriptor.

Also when creating a view to a texture, it requires a `format` field in the descriptor.

Requiring both seems redundant. The format of a view should establish the interpretation of the texture? I.e. a texture is an uninterpreted stream of bits per pixel, and a view is supposed to define the interpretation of the pixel data? The `format` of a texture itself is then redundant.

WebGPU does not currently have "unsized" texture formats like D3D has, so a texture cannot be created with e.g. a `R32_TYPELESS` format.

What is the relationship between these two format fields?

Does it matter what one specifies as a format for a texture, except for the purposes of establishing the bits-per-pixel allocation? (and/or flagging a special depth/stencil/block format?)

Tried looking for an explainer in the spec, but was not able to find a relevant paragraph.

Should GPUTexture `format` field be removed in favor of a typeless format field, or a general bitsPerPixel field?

**Comment by Kangz** [comment, marked as answer]
The `format` is really necessary for the `GPUTextureDescriptor` because a lot of things in the drivers are done based on the format (and not just the bitsPerPixel).

The `format` in `GPUTextureViewDescriptor` is optional (otherwise it would have `required` in from of it in the IDL) and default to the texture's format. When using WebGPU that's done with the sentinel value of `WGPUTextureFormat_Undefined` that's 0 and the default initialized value in C.

The reason why there is a format for the view is if you want to do "format reinterpretation" between compatible formats (they have to have the same bitsPerTexel but also more). This feature was heavily discussed in #744 and #168 and IIRC we decided limited format reinterpretation is something we could defer post-MVP. `GPUTextureViewDescriptor.format` is a remnant of that for when we get back to this feature.
